### PR TITLE
Beta commit

### PR DESCRIPTION
**Pull request by kentgottschalk**
Exported features in Ding Seo Metatag Content module.
Added custom tokens.

### CODE REVIEW

File: modules/ding_seo/ding_seo.tokens.inc [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+
+    // Language
+    $node['language'] = [
+        'name' => t("language"),
+        'description' => t('language'),
+    ];
+
+    // Returns the new token type as well as the tokens
+    return [
+        'types' => ['dingseotoken' => $type],
```
Comment (holt83) [inline modules/ding_seo/ding_seo.tokens.inc]: I'm defining several ding_seo specific token types, so would it be possible to define your token type something like this to avoid confusion:

```
  $info['types']['ding_seo_node_tokens'] = [
    'name' => t('Ding SEO node tokens'),
    'description' => t('Special node tokens defined by Ding SEO for use in structured data'),
    'needs-data' => 'node',
  ]; 
```
(also note the 'needs-data' option)

Otherwise I think it will be very confusing where to find the different tokens :)

File: modules/ding_seo/ding_seo.tokens.inc [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+        'name' => t("Event Enddate"),
+        'description' => t('Event enddate in YYYY-MM-DD HH:MM::SS format'),
+    ];
+    $node['event_free'] = [
+        'name' => t("Event Free Boolean"),
+        'description' => t('Boolean of whether the event is free or not '),
+    ];
+
+    // Language
+    $node['language'] = [
```
Comment (holt83) [inline modules/ding_seo/ding_seo.tokens.inc]: Is there really no language token defined by default on node? 

It appears that [node defines a language token](https://api.drupal.org/api/drupal/modules%21node%21node.tokens.inc/function/node_token_info/7.x), but maybe that can't be used? Perhaps a comment about this?

File: modules/ding_seo/features/ding_seo_metatag_content/ding_seo_metatag_content.features.metatag.inc [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+ */
+
+/**
+ * Implements hook_metatag_export_default().
+ */
+function ding_seo_metatag_content_metatag_export_default() {
+  $config = array();
+
+  // Exported Metatag config instance: node.
+  $config['node'] = array(
```
Comment (holt83) [inline modules/ding_seo/features/ding_seo_metatag_content/ding_seo_metatag_content.features.metatag.inc:14]: This is the default global config, so no need to export. I also believe we have the title-tag configured elsewhere so it will be overridden. 

Maybe it would be a good idea to move some of the config from panels to here, but I think we should concentrate on the schema related stuff here.

I will do manual merge and fix the export, so no need to change.

File: modules/ding_seo/ding_seo.module [hunk 4]
```diff
@@ -4,3 +4,6 @@
  * @file
  * Ding SEO module file.
  */
+
+// includes custom ding_seo tokens
+include_once 'ding_seo.tokens.inc';
```
Comment (holt83) [inline modules/ding_seo/ding_seo.module]: This file should be included automatically when token related hooks are invoked.

**Comment by holt83** [comment]
Looking good. I just have a few comments.

I will do a manual merge and fix the features export.

Do you have time to look into the comments before meeting? Especially this one: https://github.com/vejlebib/ding2/pull/8#discussion_r358015380

File: modules/ding_seo/ding_seo.tokens.inc (hunk 1 above)
Comment (holt83) [reply modules/ding_seo/ding_seo.tokens.inc]: Another thing. If you look at my [`hook_token_info()` implementation](https://github.com/vejlebib/ding2/pull/7/files#diff-f6e47321f5e4c50d96d43b297bfbe4bfR17), I add the token types to the ting_object token and not as a "Global token".

I think it has the advantage of making it more clear that this token only works on the node token and the "needs-data" enforces this.

If you don't have "needs-data" wont these tokens just show up like you can use them everywhere, even when there's no node context?

File: modules/ding_seo/ding_seo.module (hunk 4 above)
Comment (kentgottschalk) [reply modules/ding_seo/ding_seo.module]: Will remove this as suggested.

File: modules/ding_seo/ding_seo.tokens.inc (hunk 2 above)
Comment (kentgottschalk) [reply modules/ding_seo/ding_seo.tokens.inc]: The language token is there to set language und as dk, as there is a lot of nodes with undefined language. 

case 'language':
                    if ($data["node"]->language == "und") {
                        $replacements[$original] = "dk";
                    } else {
                        $replacements[$original] = $data["node"]->language;
                    }
                    break;

Comment (holt83) [reply modules/ding_seo/ding_seo.tokens.inc]: But wont this override the language token defined by the node module and perhaps cause unexpected behavior elsewhere?

File: modules/ding_seo/features/ding_seo_metatag_content/ding_seo_metatag_content.features.metatag.inc (hunk 3 above)
Comment (kentgottschalk) [reply modules/ding_seo/features/ding_seo_metatag_content/ding_seo_metatag_content.features.metatag.inc:14]: Great if you'll do a manuelt merge.

Comment (holt83) [reply modules/ding_seo/features/ding_seo_metatag_content/ding_seo_metatag_content.features.metatag.inc:14]: Is there a reason for having the features in a separate module? Otherwise I will move them into the main module ding_seo.

File: modules/ding_seo/ding_seo.info [hunk 5]
```diff
@@ -0,0 +1,4 @@
+name = Ding SEO
+description = Provides various SEO functionality to ding2
+package = Ding!
+core = "7.x"
```
Comment (holt83) [inline modules/ding_seo/ding_seo.info:4]: The info file and install file is added in the main ding-seo branch, so just remove this before rebasing.

File: modules/ding_seo/ding_seo.install [hunk 6]
```diff
@@ -0,0 +1,6 @@
+<?php
+
+/**
+ * @file
+ * Ding SEO install file.
+ */
```
Comment (holt83) [inline modules/ding_seo/ding_seo.install:6]: The info file and install file is added in the main ding-seo branch, so just remove this before rebasing.

File: modules/ding_seo/ding_seo.tokens.inc (hunk 2 above)
Comment (holt83) [reply modules/ding_seo/ding_seo.tokens.inc]: Nevermind above comment: If we make a `ding_seo_node_tokens` group it wont be an issue.

File: modules/ding_seo/features/ding_seo_metatag_content/ding_seo_metatag_content.features.metatag.inc (hunk 3 above)
Comment (kentgottschalk) [reply modules/ding_seo/features/ding_seo_metatag_content/ding_seo_metatag_content.features.metatag.inc:14]: No other reason other than it's the way we usually do it in Odense, keeping the features and code separate, so it's perfectly fine to move it into main module.

Comment (holt83) [reply modules/ding_seo/features/ding_seo_metatag_content/ding_seo_metatag_content.features.metatag.inc:14]: I don't mind doing having features in a separate module if there's some good reason for it? The most important thing is that we're consistent :)

File: modules/ding_seo/ding_seo.tokens.inc (hunk 1 above)
Comment (kentgottschalk) [reply modules/ding_seo/ding_seo.tokens.inc]: I've rewritten the token functions to use node & needs data.

File: modules/ding_seo/ding_seo.tokens.inc (hunk 2 above)
Comment (kentgottschalk) [reply modules/ding_seo/ding_seo.tokens.inc]: Ok

File: modules/ding_seo/features/ding_seo_metatag_content/ding_seo_metatag_content.features.metatag.inc (hunk 3 above)
Comment (kentgottschalk) [reply modules/ding_seo/features/ding_seo_metatag_content/ding_seo_metatag_content.features.metatag.inc:14]: It's fine if you move it into the main module.

File: .gitignore [hunk 7]
```diff
@@ -6,3 +6,4 @@ themes/ddbasic/scripts/min/
 config.php
 tests/behat/.idea/
 .gitignore
+/cygdrive
```
Comment (holt83) [inline .gitignore:9]: This change appears to be unrelated to the PR?

**Comment by holt83** [comment]
@kentgottschalk 

It appears that some unrelated changes has sneaked its way in: https://github.com/vejlebib/ding2/pull/8/files#r368984898 

Also the reason for the merge conflicts is that you have duplicated my initial commit from the ding-seo branch: https://github.com/vejlebib/ding2/pull/8/commits/7e018f01fe6f6a69907c80dc68e07a6cf69f3ee6

Can you make a new rebase where you exclude that commit?

**Comment by holt83** [comment]
Manually rebased and merged to main ding-seo branch.
